core/controllers/topic_editor.py

```diff
@@ -251,3 +285,33 @@ def put(self, topic_id, assignee_id):
         })
 
         self.render_json(self.values)
+
+
+class TopicPublishHandler(base.BaseHandler):
+    """A handler for publishing and unpublishing topics."""
+
+    @acl_decorators.can_manage_rights_for_topic
+    def put(self, topic_id):
+        """Publishes or unpublishes a topic.
+        """
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        publish_status = self.payload.get('publish_status')
+
+        if not isinstance(publish_status, bool):
+            raise self.InvalidInputException(
+                Exception('Publish status should only be true or false.'))
+
+        try:
+            if publish_status:
+                topic_services.publish_topic(topic_id, self.user_id)
+            else:
+                topic_services.unpublish_topic(topic_id, self.user_id)
+        except Exception as e:
+            raise self.UnauthorizedUserException(e)
+
+        self.values.update({
+            'topic_status_updated': True
```

Comment:
Is this actually needed? Wouldn't a 200 status code mean that everything worked fine?

Reply:
Yeah, done.